### ISSUE TITLE
Adjusting tag cardinality can cause duplicate metrics

### ISSUE THREAD

**Issue by spacez320** [DataDog/datadog-agent#5904]
**Describe what happened:**

We've seen it be the case that when we adjust Tag cardinality (adding a new Tag, for example), two metric series which logically represent the same data can coexist for a short period of time (probably the length of time to collect new data) and cause counting issues in Monitors or Dashboards.

For example, I have some metric in the Kubernetes integration `kubernetes.node.whatever`, and I want to add a new `nodeLabelAsTag` setting `foo=bar`. I push that configuration change to the Agent, and the following happens:

Before push, metrics look like this:
```
kubernetes.node.whatever{fizz=buzz} 1
```

After push, they look like this for a few minutes:
```
kubernetes.node.whatever{fizz=buzz} 1
kubernetes.node.whatever{fizz=buzz, foo=bar} 1
```

Then eventually they become:
```
kubernetes.node.whatever{fizz=buzz, foo=bar} 1
```

The problem with this is that anything that tries to `sum:kubernetes.node.whatever{fizz=buzz}` will temporarily see "2" when really it should be "1".

**Describe what you expected:**

I'm not sure what's possible, but it would be nice if we could teach Monitor or Datadog metrics to deduplicate accessory cardinality. Currently it's the case that changes like this can cause our Monitors to go haywire, and we would like to edit Tags while avoiding this.

**Steps to reproduce the issue:**

1. Create a metric.
2. Create a Monitor that sums the metric on an existing Tag.
3. Add a Tag on that metric.
4. Observe that the Monitor tracking value jumps because it's seeing more data.

**Additional environment details (Operating System, Cloud provider, etc):**

Seems like a pretty general problem.

**Comment by programmer04** [comment]
Hello, @spacez320 I faced a similar issue, I wanted to discard some tags to avoid duplication of metrics that represent the same logical value (I wanted to get rid of tag `host`, because the metric was service level, but was reported by every replica). `Distributions` metric type helps me because it allows select tags to retain. You can read more https://docs.datadoghq.com/metrics/distributions/

**Comment by BEvgeniyS** [comment]
We have same problem with `kubernetes.memory.request` is doubled for each new value of `container_id` (for example if container is restarted). It takes way too long for agent to remove old container_id tag, and during that period, the sum metric is doubled

**Comment by spacez320** [comment]
rip

**Comment by chouetz** [comment]
Hi @spacez320,

Thank you for taking the time to share your feedback. We recently introduced a system to help manage long-standing issues through automated stale management. We understand this may have come as a surprise for some existing issues. We want to apologize if it felt that way. It was not our intent, and we want to reassure you that every issue is important to us and reviewed carefully. For your specific issue, we’re reviewing it to ensure the discussion progresses and receives the right level of attention. Thank you for helping improve the Datadog Agent.


— The Datadog Agent Team

**Comment by Mathew-Estafanous** [comment]
Hi. Thanks for bringing this up.

The "duplicate" metrics observed during tag cardinality changes are not actually two duplicate metrics sent by the Agent, but rather an artifact of how the backend handles time series continuity. Instead, what you're seeing is the backend interpolating the old series forward just as the new series (with the new tags) begins. This overlap causes the sum aggregation to temporarily double count.

The best way to prevent this is to use a [Fill (null) Modifier](https://docs.datadoghq.com/metrics/guide/interpolation-the-fill-modifier-explained/) to your query. This stops the backend from inferring values for the old series once it stops reporting, ensuring a clean hand-off to the new tags without the spike.

Example:
```
sum:kubernetes.node.whatever{fizz=buzz}.fill(null)
```

**TLDR;** Add .fill(null) to ensure that as soon as the old tag set stops reporting, it is treated as null rather than carrying its last known value forward into the sum.
